core/src/input_reader/guest_input_reader.rs

```diff
@@ -485,10 +551,21 @@ fn get_balance_churn_limit(spec: &ChainSpec, total_active_balance: u64) -> Resul
     churn.safe_sub(churn.safe_rem(spec.effective_balance_increment)?)
 }
 
-/// Extracts an u64 from a 32-byte SSZ chunk.
-fn u64_from_chunk(node: &[u8; 32]) -> u64 {
-    assert!(node[8..].iter().all(|&b| b == 0));
-    u64::from_le_bytes(node[..8].try_into().unwrap())
+trait U64ChunkPos {
+    fn as_range() -> std::ops::Range<usize>;
+}
+
+struct Leftmost;
+
+impl U64ChunkPos for Leftmost {
+    fn as_range() -> std::ops::Range<usize> {
+        0..8
+    }
+}
+
+/// Extracts an u64 from a 32-byte SSZ chunk
+fn u64_from_chunk<P: U64ChunkPos>(node: &[u8; 32]) -> u64 {
+    u64::from_le_bytes(node[P::as_range()].try_into().unwrap())
```

Comment:
Previously, we checked whether the remaining bytes are all zeroes.
I believe we should absolutely keep that. I don't think it's vulnerable otherwise. However, if we accidentally try to read a u128 as a u64, for example, this change would detect it.

Reply:
See above. If we check for all zeroes and try and read the block slot that would error due to it being packed with the `proposer_index`

This only worked before due the the packing rule always being applied